sympy/ntheory/generate.py

```diff
@@ -449,9 +449,10 @@ def eval(cls, n):
         if n is S.Infinity:
             return S.Infinity
 
-        if n.is_Integer:
+        if n.is_real and n.is_number:
```

Comment:
`int(n)` should work for real numbers. Would it suffice to just try `n = int(n)` and return None in case of TypeError?

Reply:
Sure, if you think that's better. Should I also try to raise a ValueError for complex values, since right now you can call primepi(1 + I) (what's the correct way to do that?)

Reply:
Perhaps `n.is_real` could be checked in the TypeError case and ValueError raised if that is false.
There might be additional cases that could be checked even before trying `int(n)`: `-oo`, `zoo`, `nan` (maybe ValueError?).